hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/security/S3GetSecretResponse.java

```diff
@@ -58,13 +59,10 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
     boolean isOk
         = getOMResponse().getStatus() == OzoneManagerProtocolProtos.Status.OK;
     if (s3SecretValue != null && isOk) {
-      if (s3SecretManager.isBatchSupported()) {
-        s3SecretManager.batcher().addWithBatch(batchOperation,
-            s3SecretValue.getKerberosID(), s3SecretValue);
-      } else {
-        s3SecretManager.storeSecret(s3SecretValue.getKerberosID(),
-            s3SecretValue);
-      }
+      S3Batcher batcher = s3SecretManager.batcher();
+      if (batcher != null) {
```

Comment:
Why is the condition changed?  The `else` block moved to `S3GetSecretRequest` continues to check `isBatchSupported()`.  This might introduce a mismatch if `S3SecretManager` implementation deviates from the default `isBatchSupported()` implementation.

Reply:
@Cyrill can you please let me know if the condition is intentionally changed?

From:

```java
if (s3SecretManager.isBatchSupported())
```

to:

```java
if (batcher != null)
```

If not, can we restore the original one?

Other than that, I think this patch is good to go.